rllib/core/models/configs.py

```diff
@@ -0,0 +1,233 @@
+
+            return TorchMLPHead(self)
+        else:
+            from ray.rllib.core.models.tf.mlp import TfMLPHead
+
+            return TfMLPHead(self)
+
+
+@ExperimentalAPI
+@dataclass
+class MLPEncoderConfig(MLPHeadConfig):
+    """Configuration for an MLP that acts as an encoder.
+
+    Although it inherits from MLPHeadConfig, it does not output an MLPHead.
+    This inheritance is solely to unify the configuration options between MLPEncoders
+    and MLPHeads.
+
+    See ModelConfig for usage details.
+    """
+
+    @_framework_implemented()
+    def build(self, framework: str = "torch") -> Encoder:
+        # Activation functions in TF are lower case
+        self.output_activation = _convert_to_lower_case_if_tf(
+            self.output_activation, framework
+        )
+        self.hidden_layer_activation = _convert_to_lower_case_if_tf(
+            self.hidden_layer_activation, framework
+        )
+
+        if framework == "torch":
+            from ray.rllib.core.models.torch.encoder import TorchMLPEncoder
+
+            return TorchMLPEncoder(self)
+        else:
+            from ray.rllib.core.models.tf.encoder import TfMLPEncoder
+
+            return TfMLPEncoder(self)
+
+
+@ExperimentalAPI
+@dataclass
+class LSTMEncoderConfig(ModelConfig):
+    """Configuration for a LSTM encoder.
+
+    See ModelConfig for usage details.
+
+    Attributes:
+        input_dim: The input dimension of the network. It cannot be None.
+        hidden_dim: The size of the hidden layer.
+        num_layers: The number of LSTM layers.
+        batch_first: Wether the input is batch first or not.
+        output_activation: The activation function to use for the output layer.
+        observation_space: The observation space of the environment.
+        action_space: The action space of the environment.
+        view_requirements_dict: The view requirements to use if anything else than
+            observation_space or action_space is to be encoded. This signifies an
+            advanced use case.
+        get_tokenizer_config: A callable that takes a gym.Space and a dict and
+            returns a ModelConfig to build tokenizers for observations, actions and
+            other spaces that might be present in the view_requirements_dict.
+
+    """
+
+    input_dim: int = None
+    hidden_dim: int = None
+    num_layers: int = None
+    batch_first: bool = True
+    output_activation: str = "linear"
+    observation_space: gym.Space = None
+    action_space: gym.Space = None
+    view_requirements_dict: ViewRequirementsDict = None
+    get_tokenizer_config: Callable[[gym.Space, Dict], ModelConfig] = None
+
+    @_framework_implemented(tf2=False)
+    def build(self, framework: str = "torch") -> Encoder:
+        if framework == "torch":
+            from ray.rllib.core.models.torch.encoder import TorchLSTMEncoder
+
+            return TorchLSTMEncoder(self)
+
+
+@ExperimentalAPI
+@dataclass
+class IdentityEncoderConfig(ModelConfig):
```

Comment:
I still think we should get rid of this Identity encoder in this PR, it won't be used in any of the upcoming PRs, right?

Reply:
Would catalog use this in any way?

Reply:
Nope, I'm getting rid of it.